tests/integration/tests/graph_nodes_test.go

```diff
@@ -0,0 +1,82 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/kiali/kiali/tests/integration/utils"
+)
+
+func TestAppGraph(t *testing.T) {
+	assert := assert.New(t)
+	name := "details"
+	graphType := "app"
+	assertGraphConfig("applications", graphType, utils.BOOKINFO, name, assert)
+}
+
+func TestVersionedAppGraph(t *testing.T) {
+	assert := assert.New(t)
+	name := "details"
+	graphType := "versionedApp"
+	assertGraphConfig("applications", graphType, utils.BOOKINFO, name, assert)
+}
+
+func TestAppGraphEmpty(t *testing.T) {
+	assert := assert.New(t)
+	name := "detailswrong"
+	graphType := "app"
+	assertEmptyGraphConfig("applications", graphType, utils.BOOKINFO, name, assert)
+}
+
+func TestServiceGraph(t *testing.T) {
+	assert := assert.New(t)
+	name := "details"
+	graphType := "versionedApp"
+	assertGraphConfig("services", graphType, utils.BOOKINFO, name, assert)
+}
+
+func TestServiceGraphEmpty(t *testing.T) {
+	assert := assert.New(t)
+	name := "detailswrong"
+	graphType := "workload"
+	assertEmptyGraphConfig("services", graphType, utils.BOOKINFO, name, assert)
+}
+
+func TestWorkloadGraph(t *testing.T) {
+	assert := assert.New(t)
+	name := "reviews-v2"
+	graphType := "workload"
+	assertGraphConfig("workloads", graphType, utils.BOOKINFO, name, assert)
+}
+
+func TestWorkloadGraphEmpty(t *testing.T) {
+	assert := assert.New(t)
+	name := "reviews-wrong"
+	graphType := "workload"
+	assertEmptyGraphConfig("workloads", graphType, utils.BOOKINFO, name, assert)
+}
+
+func assertGraphConfig(objectType, graphType, namespace, name string, assert *assert.Assertions) {
+	graph, _, _ := utils.Graph(map[string]string{"graphType": graphType, "namespaces": namespace})
+	// TODO better way to check if there are any graph nodes at all to be able to verify requested ones
+	if len(graph.Elements.Nodes) > 0 && len(graph.Elements.Edges) > 0 {
```

Comment:
If the graph is empty, should we fail here instead of skipping the asserts when `len(Nodes) && len(Edges) == 0`?

Reply:
yes, added wait there